### PR TITLE
Teacher updates student id

### PR DESCRIPTION
**Pull request by nathanstitt**
Goes with https://github.com/openstax/tutor-js/pull/1288, can be merged before it is though

### CODE REVIEW

File: app/representers/api/v1/student_teacher_update_representer.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    property :student_identifier,
+             type: String,
+             writeable: true,
+             readable: true,
+             schema_info: {
+               required: true
+             }
+
+  end
+end
```
Comment (jpslav) [inline app/representers/api/v1/student_teacher_update_representer.rb]: Why not just put this in the existing `StudentRepresenter`?

File: app/controllers/api/v1/students_controller.rb [hunk 2]
```diff
@@ -37,18 +37,23 @@ def update_self
     #{json_schema(Api::V1::StudentRepresenter, include: :writeable)}
   EOS
   def update
+    @period_update_result = nil
     @student.with_lock do
       OSU::AccessPolicy.require_action_allowed!(:update, current_api_user, @student)
-      payload = consume!(Hashie::Mash.new, represent_with: Api::V1::StudentRepresenter)
-      period = CourseMembership::Models::Period.find(payload['course_membership_period_id'])
-      @result = MoveStudent.call(student: @student, period: period)
-      OSU::AccessPolicy.require_action_allowed!(:update, current_api_user, @student)
```
Comment (jpslav) [inline app/controllers/api/v1/students_controller.rb]: Why was this line here a second time, and why is it gone now?

File: app/representers/api/v1/student_teacher_update_representer.rb (hunk 1 above)
Comment (nathanstitt) [inline app/representers/api/v1/student_teacher_update_representer.rb]: While they're identical right now, I thought that the Teacher shouldn't be able to update the same fields the student does.  

For instance in the future we might allow a student to directly edit their name (w/out going through accounts), but we wouldn't want a Teacher to do so.

File: app/controllers/api/v1/students_controller.rb (hunk 2 above)
Comment (nathanstitt) [inline app/controllers/api/v1/students_controller.rb]: Hm, at first I thought it was an oversight, but now suspect it's checking that teacher also has permissions on the period the student is moving to.  It uses `UserIsCourseTeacher` to verify the user is a teacher on the student's course.  I'm restoring it and commenting to that effect
